todoApp/spec/main.tsp

```diff
@@ -250,13 +245,21 @@ namespace TodoItems {
 
   @list op list(...PaginationControls): WithStandardErrors<TodoPage>;
 
+  @sharedRoute
   @post
-  op create(
+  op createJson(
     @header contentType: "application/json",
     item: TodoItem,
     attachments?: TodoAttachment[],
   ): WithStandardErrors<TodoItem | InvalidTodoItem>;
 
+  @sharedRoute
+  @post
+  op createForm(
+    @header contentType: "multipart/form-data",
+    @body body: ToDoItemMultipartRequest,
```

Comment:
`@multipartBody`?

Reply:
@markcowl Since language SDK including Python/Java doesn't support union mixed File  with common model in multipart request body, we hope to abandon Union in multipart request like:
```
  @sharedRoute
  @post
  op createFormFile(
     @header contentType: "multipart/form-data",
     @multipartBody body: {item: HttpPart<TodoItem>; attachments?: HttpPart<TodoFileAttachment>[];},
  ): WithStandardErrors<TodoItem | InvalidTodoItem>;

  @sharedRoute
  @post
  op createFormUrl(
     @header contentType: "multipart/form-data",
     @multipartBody body: {item: HttpPart<TodoItem>; attachments?: HttpPart<TodoUrlAttachment>[];},
  ): WithStandardErrors<TodoItem | InvalidTodoItem>;
```

Reply:
These are semantically different though, and would disallow mixing url attachments with File attachments.

Reply:
But for this version of the demo, we should make sure that the API is well -supported, so we cna simplify this

Reply:
@markcowl 
`I think we might limit the multipart attachment to only files, and allow files or Urls through the json endpoint`
-> That makes sense and we can update https://github.com/allenjzhang/typespec-e2e-demo/blob/052143e392c541fd03a2eaffd7a80f689b499720/todoApp/spec/main.tsp#L82-L85 to 
```
model ToDoItemMultipartRequest {
  item: HttpPart<TodoItem>;
  attachments?: HttpPart<TodoFileAttachment>[];
}
``